test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -0,0 +1,1083 @@
+        {
+            "enabled"         : true,
+            "default"         : "256MB",
+            "reserve_amount"  : "50MB",
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : ["no_cgroups_mem"]
+        },
+        "memsw":
+        {
+            "enabled"         : true,
+            "default"         : "256MB",
+            "reserve_amount"  : "45MB",
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : []
+        }
+    }
+}
+"""
+        self.cfg4 = """{
+    "cgroup_prefix"         : "pbspro",
+    "exclude_hosts"         : [],
+    "exclude_vntypes"       : ["no_cgroups"],
+    "run_only_on_hosts"     : [],
+    "periodic_resc_update"  : true,
+    "vnode_per_numa_node"   : false,
+    "online_offlined_nodes" : true,
+    "use_hyperthreads"      : false,
+    "cgroup":
+    {
+        "cpuacct":
+        {
+            "enabled"         : true,
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : []
+        },
+        "cpuset":
+        {
+            "enabled"         : true,
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : ["no_cgroups_cpus"]
+        },
+        "devices":
+        {
+            "enabled"         : false
+        },
+        "hugetlb":
+        {
+            "enabled"         : false
+        },
+        "memory":
+        {
+            "enabled"         : true,
+            "default"         : "256MB",
+            "reserve_amount"  : "100MB",
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : ["no_cgroups_mem"]
+        },
+        "memsw":
+        {
+            "enabled"         : true,
+            "default"         : "256MB",
+            "reserve_amount"  : "90MB",
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : []
+        }
+    }
+}
+"""
+        Job.dflt_attributes[ATTR_k] = 'oe'
+        # Increase the log level
+        a = {'log_events': '2047'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+        # Configure the scheduler to schedule using vmem
+        a = {'resources': 'ncpus,mem,vmem,host,vnode'}
+        self.scheduler.set_sched_config(a)
+        # Configure the mom
+        self.momA = self.moms.values()[0]
+        c = {'$logevent': '0xffffffff', '$clienthost': self.server.name,
+             '$min_check_poll': 8, '$max_check_poll': 12}
+        self.momA.add_config(c)
+        # Create resource as root
+        attr = {'type': 'long', 'flag': 'nh'}
+        rc = self.server.manager(MGR_CMD_CREATE, RSC, attr, id='nmics',
+                                 runas='root', logerr=False)
+        rc = self.server.manager(MGR_CMD_CREATE, RSC, attr, id='ngpus',
+                                 runas='root', logerr=False)
+        # Import the hook
+        self.hook_name = 'pbs_cgroups'
+        hook_url = 'https://raw.githubusercontent.com/' + \
+            'PBSPro/pbspro/master/src/hooks/cgroups/pbs_cgroups.py'
+        self.load_hook(hook_url)
```

Comment:
I'll update this to pull the hook from PBS_EXEC/lib/python/altair/pbs_hooks/pbs_cgroups.PY